waitress/app/utils.py

```diff
@@ -128,98 +130,82 @@ def difference(cls, repo_users, db_users):
-
-        for user in repo_users:
-            if user not in users:
-                unsaved_users.append(user)
-
-        # if user is in slack group and not in database
-        for user in db_users:
-            if user not in users:
-                unsaved_users.append(user)
+        users = [user.slack_id for user in db_users]
 
-        return unsaved_users
+        # return all users from slack that don't exist in the db
+        return [user for user in repo_users if user not in users]
 
     @classmethod
     def filter_add_user(cls, info, difference, trim):
         """
         A method that retrieves users' info using the difference.
         """
+        def is_user_invalid(user):
+            is_deleted = user.get('deleted')
+            is_bot = user.get('is_bot')
+            email = user.get('profile').get('email', '')
+            is_email_valid = email.endswith(settings.DOMAIN_LIST)
+
+            return (is_deleted or is_bot or (not is_email_valid))
+
         def normalize(info):
             """
             A function that normalizes retrieved information from Slack.
             """
-            user_dict = {}
-            not_user_list = []
+            valid_users = {}
+            invalid_users = []
 
             for item in info:
-                if 'deleted' in item and item['deleted'] is True:
-                    not_user_list.append(item['id'])
-                    continue
-                if 'is_bot' in item and item['is_bot'] is True:
-                    not_user_list.append(item['id'])
-                    continue
-                if 'email' not in item['profile'] or item['profile'].get('email') is None:
-                    not_user_list.append(item['id'])
-                    continue
-
-                if not item['profile']['email'].endswith(settings.DOMAIN_LIST):
-                    not_user_list.append(item['id'])
-                    continue
-
-                firstname = item.get('profile').get('first_name', '')
-                lastname = item.get('profile').get('last_name', '')
-                user_dict[item['id']] = {
-                    'slack_id': item['id'],
-                    'email': item['profile']['email'],
-                    # use slack default image
-                    'photo': item['profile'].get('image_original', item['profile'].get('image_192')),
-                    'firstname': firstname.title(),
-                    'lastname': lastname.title()
-                }
-
-            return user_dict, not_user_list
-
-        info, not_user = normalize(info)
-
-        difference = cls.get_real_users(difference[:], not_user)
-
-        if len(difference) == 0:
-            return "Users list not changed"
+                user_slack_id = item['id']
+
+                if is_user_invalid(item):
+                    invalid_users.append(user_slack_id)
+                else:
+                    firstname = item.get('profile').get('first_name', '')
```

Comment:
This is a dictionary comprehension


```
def gooddict(id, profile):
   return {'slack_id', id,
                email: profile.get('email')
        .....
}
valid_users = {item[id]: goodidct(item.get('profile') for item in info if not is_invalid_user(item)}
```
Somehow I believe this should be any array as first instead of first converting it to a dictionary

so it can just be 
```
valid_slack_users = [gooddict(item.get('profile) for for item in info if not is_invalid_user(item)]
```
no need for the indirections of valid_users first or invalid users that wasn't used